### PR TITLE
[Sil][User][UserBundle] Remove Sylius and Sonata for User Management

### PR DESCRIPTION
**Pull request by FanchTheSystem**
### Summary

Would like to stop depending on third party tools for user management

### Other Information

For our scrum users' stories :)

### CODE REVIEW

File: app/config/security.yml [hunk 1]
```diff
@@ -14,19 +13,16 @@ security:
...
 
         ROLE_SUPER_ADMIN:
-            - ROLE_SONATA_ADMIN
-            - ROLE_ALLOWED_TO_SWITCH
+            - ROLE_ADMIN
 
+    # todo use with bcrypt
     encoders:
-        Sylius\Component\User\Model\UserInterface: sha512
+        Sil\Bundle\UserBundle\Entity\User: plaintext
```
Comment (RomainSanchez) [inline app/config/security.yml]: "plaintext"  :gun:

File: app/config/security.yml [hunk 2]
```diff
@@ -14,19 +13,16 @@ security:
...
 
     providers:
-        sonata_user_provider:
-            id: sylius.sonata_user_provider.email_or_name_based
-        sylius_admin_user_provider:
-            id: sylius.admin_user_provider.email_or_name_based
-        sylius_shop_user_provider:
-            id: sylius.shop_user_provider.email_or_name_based
+        sil_user_provider:
```
Comment (RomainSanchez) [inline app/config/security.yml]: sil_user.provider.user or maybe sil_user.user_provider ?

File: src/Sil/Bundle/UserBundle/Controller/UserController.php [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+use Sil\Component\User\Repository\UserRepositoryInterface; //Sil\Bundle\UserBundle\Repository\UserRepositoryInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Sil\Bundle\UserBundle\Form\Type\CreateUserType;
+use Sil\Bundle\UserBundle\Form\Type\EditUserType;
+
+/* Ah Ah copy past from Product Bundle */
```
Comment (RomainSanchez) [inline src/Sil/Bundle/UserBundle/Controller/UserController.php]: ...

File: src/Sil/Bundle/UserBundle/Controller/UserController.php [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+     * @return Response
+     */
+    public function listAction(Request $request): Response
+    {
+        $this->breadcrumbBuilder
+            ->addItemWithRoute(
+                $this->get('translator')->trans('sil.user.user.list.page_header.title'),
+                'sil_user_list'
+            )
+            ;
```
Comment (RomainSanchez) [inline src/Sil/Bundle/UserBundle/Controller/UserController.php]: indentation

File: src/Sil/Bundle/UserBundle/Entity/User.php [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+ */
+
+namespace Sil\Bundle\UserBundle\Entity;
+
+//use Doctrine\Common\Collections\ArrayCollection;
+//use Doctrine\Common\Collections\Collection;
+use Sil\Component\User\Model\User as SilUser;
+use Blast\Bundle\BaseEntitiesBundle\Entity\Traits\Guidable;
+
+/* Sql for test:
```
Comment (RomainSanchez) [inline src/Sil/Bundle/UserBundle/Entity/User.php]: ???

File: src/Sil/Bundle/UserBundle/Form/Transformer/ArrayToUserTransformer.php [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+                    $value,
+                    $this->mandatoryFields
+                );
+            }
+            $user = new $this->userClass();
+            $user->setUsername($value['username']);
+            $user->setPassword($value['password']);
+            $user->setEmail($value['email']);
+        }
+        $user->setEnabled(true); /* @todo use a tock to enable or disable it */
```
Comment (RomainSanchez) [inline src/Sil/Bundle/UserBundle/Form/Transformer/ArrayToUserTransformer.php]: remove comment

File: src/Sil/Bundle/UserBundle/Form/Type/CreateUserType.php [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+namespace Sil\Bundle\UserBundle\Form\Type;
+
+use Sil\Bundle\UserBundle\Form\Transformer\ArrayToUserTransformer;
+use Symfony\Component\Form\Extension\Core\Type\BaseType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+/* Ah Ah copy past from Product Bundle */
```
Comment (RomainSanchez) [inline src/Sil/Bundle/UserBundle/Form/Type/CreateUserType.php]: ....

File: src/Sil/Bundle/UserBundle/Resources/config/services.yml [hunk 8]
```diff
@@ -0,0 +1,45 @@
+services:
+
+    ### REPOSITORY
+
+    sil.repository.user:
```
Comment (RomainSanchez) [inline src/Sil/Bundle/UserBundle/Resources/config/services.yml]: sil_user.repository.user

File: src/Sil/Bundle/UserBundle/Entity/Traits/Nameable.php [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+ * This file is licenced under the GNU LGPL v3.
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Sil\Bundle\UserBundle\Entity\Traits;
+
+/* @todo: remove this or not or use this */
+
+trait Nameable
```
Comment (RomainSanchez) [inline src/Sil/Bundle/UserBundle/Entity/Traits/Nameable.php]: Use Nameable trait from blastBaseEntitiesBundle instead

File: app/config/security.yml (hunk 1 above)
Comment (FanchTheSystem) [reply app/config/security.yml]: It is easier for test to have plaintext, bcrypt will be enable in another user story (6327)

File: src/Sil/Bundle/UserBundle/Entity/Traits/Nameable.php (hunk 9 above)
Comment (FanchTheSystem) [reply src/Sil/Bundle/UserBundle/Entity/Traits/Nameable.php]: Not the same usage, as here is for firstname and lastname, the one from blast is from for name only

**Pull request merged**